### PR TITLE
Add Python implementation of the `PointCloudDataFrame` class

### PR DESCRIPTION
**Pull request by jp-dark**
Add the Python implementation of the new `PointCloudDataFrame` class.

**Issue:**  #3046 #3108

### CODE REVIEW

File: apis/python/tests/test_coordinates.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import tiledbsoma as soma
+
+
+def check_transform_is_equal(
+    actual: CoordinateTransform, desired: CoordinateTransform
+) -> None:
+    assert actual.input_axes == desired.input_axes
+    assert actual.output_axes == desired.output_axes
+    if isinstance(desired, IdentityTransform):
```
Comment (johnkerl) [inline apis/python/tests/test_coordinates.py:19]: nit for the future: `expected` is a more standard wording than `desired` (although the latter is more fresh & creative :) )

**Review by johnkerl — APPROVED** [review]

File: apis/python/tests/test_coordinates.py (hunk 1 above)
Comment (jp-dark) [reply apis/python/tests/test_coordinates.py:19]: I can't take the credit - I store the name from [numpy](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html)

Comment (johnkerl) [reply apis/python/tests/test_coordinates.py:19]: OK TIL, thank you! :)

File: apis/python/src/tiledbsoma/_scene.py [hunk 2]
```diff
@@ -235,14 +235,14 @@ def set_transform_to_point_cloud(
...
-        a point cloud stored in the scene.
+        a point cloud dataframe  stored in the scene.
 
         If the subcollection the point cloud is inside of is more than one
         layer deep, the input should be provided as a sequence of names. For example,
         to set a transform for  a point named `transcripts` in the `var/RNA`
         collection::
 
-            scene.set_transformation_for_point_cloud(
+            scene.set_transformation_for_point_cloud_dataframe(
```
[the remark this answers is not in the document]
Comment (jp-dark) [reply apis/python/src/tiledbsoma/_scene.py]: It's a typo.

**Pull request merged**